modules/miner/blockmanager.go

```diff
@@ -158,6 +162,14 @@ func (m *Miner) managedSubmitBlock(b types.Block) error {
 
 // SubmitHeader accepts a block header.
 func (m *Miner) SubmitHeader(bh types.BlockHeader) error {
+	if err := m.tg.Add(); err != nil {
+		return err
+	}
+	defer m.tg.Done()
+	if m.tg.IsStopped() {
+		build.Critical(errClosed)
+	}
```

Comment:
Same here, no need to both call `tg.Add()` and check `IsStopped()` - my mistake.